apps/builder/app/builder/features/breakpoints/width-input.tsx

```diff
@@ -0,0 +1,168 @@
+import { useStore } from "@nanostores/react";
+import { findApplicableMedia } from "@webstudio-is/css-engine";
+import {
+  theme,
+  Text,
+  Flex,
+  Label,
+  type NumericScrubValue,
+  InputField,
+  useId,
+  type CSS,
+  useScrub,
+} from "@webstudio-is/design-system";
+import { useCanvasWidth } from "~/builder/shared/nano-states";
+import { breakpointsStore } from "~/shared/nano-states";
+import {
+  selectedBreakpointIdStore,
+  selectedBreakpointStore,
+} from "~/shared/nano-states";
+import { useEffect, useRef, useState } from "react";
+
+// Doesn't make sense to allow resizing the canvas lower/higher than this.
+export const minWidth = 240;
+
+export const WidthInput = () => {
+  const id = useId();
+  const [canvasWidth, setCanvasWidth] = useCanvasWidth();
+  const selectedBreakpoint = useStore(selectedBreakpointStore);
+  const breakpoints = useStore(breakpointsStore);
+
+  if (canvasWidth === undefined || selectedBreakpoint === undefined) {
+    return null;
+  }
+
+  const onChange = (value: number) => {
+    setCanvasWidth(value);
+    const applicableBreakpoint = findApplicableMedia(
+      Array.from(breakpoints.values()),
+      value
+    );
+    if (applicableBreakpoint) {
+      selectedBreakpointIdStore.set(applicableBreakpoint.id);
```

Comment:
Mb doesn't do that during the edit, also from what I know stores are dumb, and better to not update if the value is already the same, to not force rerendering

Reply:
we have to do it as we scroll, breakpoints need to always reflect the size of the canvas

Reply:
I mean u need to do this 
```
if (selectedBreakpointIdStore.get() !== applicableBreakpoint.id) {
  selectedBreakpointIdStore.set(applicableBreakpoint.id);
}
```
otherwise it forces all listeners to render